jobs/pulse3d/test-pulse3d-worker/src/main.py

```diff
@@ -0,0 +1,360 @@
+                s3_client.download_file(PULSE3D_UPLOADS_BUCKET, features_parquet_key, features_filepath)
+                interactive_analysis = True
+                logger.info(f"Downloaded peaks and valleys to {features_filepath}")
+            except Exception:  # TODO catch only boto3 errors here
+                logger.info("No existing peaks and valleys found for recording")
+
+            try:
+                s3_client.download_file(PULSE3D_UPLOADS_BUCKET, zipped_recording_key, zipped_recording_path)
+                logger.info(f"Downloaded existing preanalyed data to {zipped_recording_path}")
+
+                with ZipFile(zipped_recording_path) as z:
+                    z.extractall(tmpdir)
+
+                tissue_waveforms = pl.read_parquet(tissue_waveforms_path)
+                stim_waveforms = (
+                    pl.read_parquet(stim_waveforms_path) if os.path.exists(stim_waveforms_path) else None
+                )
+
+                metadata_dict = json.load(open(metadata_path))
+                is_beta_2 = metadata_dict["file_format_version"] >= VersionInfo.parse("1.0.0")
+                existing_metadata = (
+                    MantarrayBeta2Metadata(**metadata_dict)
+                    if is_beta_2
+                    else MantarrayBeta1Metadata(**metadata_dict)
+                )
+
+                pre_analyzed_data = PreAnalyzedData(
+                    tissue_waveforms=tissue_waveforms,
+                    stim_waveforms=stim_waveforms,
+                    metadata=existing_metadata,
+                )
+
+                re_analysis = True
+            except Exception:  # TODO catch only boto3 errors here
+                logger.info(f"No existing data found for recording {zipped_recording_filename}")
+
+            # remove params that were not given as these already have default values
+            analysis_params = {k: v for k, v in metadata["analysis_params"].items() if v is not None}
+
+            if not re_analysis:
+                try:
+                    logger.info("Starting dataloader")
+                    loaded_data = from_file(recording_path)
+                except Exception:
+                    logger.exception("DataLoader failed")
+                    raise
+
+                try:
+                    logger.info("Starting preanalysis")
+                    pre_analyzed_data = process(loaded_data)
+                except UnableToConvergeError:
+                    raise Exception("Unable to converge due to low quality of data")
+                except Exception:
+                    logger.exception("PreAnalysis failed")
+                    raise
+
+            try:
+                # copy so that windowed data isn't written to S3 and used on following recordings
+                windowed_pre_analyzed_data = deepcopy(pre_analyzed_data)
+                # TODO could window stim data as well?
```

Comment:
Yeah we'll probably need to window that as well, might be nice to have pulse3d handle all the windowing so the worker doesn't need to worry about it if we need to start windowing other things in the future